### DISCUSSION TITLE
Bug correction and improvement

### DISCUSSION THREAD

**Discussion by albuquerqueraniel** [Anura3D/Anura3D_OpenSource#75] [Show and tell]
Hello, everyone!

I was trying to use absorbing boundaries combined with K0 procedure in a 2D model. When I executed the simulation, I received the following error message:

![error](https://user-images.githubusercontent.com/45863863/222923430-0b39b7a8-504c-4ee0-963d-11a5a2311fc8.png)

This error occured in the subroutine InitialiseAbsorbingBoundaryForcesForK0, which is located at MPMDynViscousBoundary.

![NodeCndVisSld](https://user-images.githubusercontent.com/45863863/222923765-98b70832-4a4d-40a9-81b7-7677f16097bb.png)

"NodeCndVisSld" isn't being allocated. Besides that, even if it was allocated, it would probably throw an error for its not receiving values. The correct variable is NodeCndVisSld2, which is populated at the subroutine CalculateNodeSurfaceAreaandSetCondSolid.

I changed the variable in InitialiseAbsorbingBoundaryForcesForK0. The One-dimensional consolidation example was used to verify if the results were right. I build a 3D model without absorbing boundaries, with K0 procedure active, and compared it with a model with absorbing boundaries. Those were the results:

![3D](https://user-images.githubusercontent.com/45863863/222924370-a193ec80-171f-4315-8109-bb9d8b24b09f.png)

When inspecting InitialiseAbsorbingBoundaryForcesForK0 subroutine I've found another problem, at least for me: The subroutine was implemented just for 3D problems. So I followed what was commented on there and expanded it to 2D problems. Here is the result of a comparision between a One-dimentional 2D model without and another with absorbing boundaries.
![2D](https://user-images.githubusercontent.com/45863863/222924744-83468fe3-4031-4515-a669-36bc6a38a108.png)

The subroutine extended for 2D is available at the following repository:

https://github.com/albuquerqueraniel/public/blob/d5be784fcc9a973f3f25fc215a3dbc29a0556a4a/InitialiseAbsorbingBoundaryForcesForK0.f

Ps.: I think there's a typo in section 3.6.3 of the TutorialManual_2022:
![material point](https://user-images.githubusercontent.com/45863863/222925790-941551a3-1dcb-4842-9d9c-e33fa894ef5c.png)
The problem has just over 200 material points, but there it tells you to choose the MP 1186.

Best regards,
Raniel

**Comment by gbtach** [comment]
Dear Raniel,

thank you very much for your contribution. We will benchmark and improve the code soon.  Regarding the manual there is indeed an issue, which is related to the fact that the geometry used for wave propagation example was actually slightly different. An element size of 0.01m and the following dimensions were used:
![image](https://user-images.githubusercontent.com/99030097/224380705-e93fcf56-d62e-4537-8cec-cb158363f0ce.png)

Thank you very much
Best regrads
Alex
